sql/catalyst/src/main/java/org/apache/spark/sql/connector/util/V2ExpressionSQLBuilder.java

```diff
@@ -103,12 +123,45 @@ protected String visitIsNotNull(String v) {
     return v + " IS NOT NULL";
   }
 
+  protected String visitStringPredicate(String name, GeneralScalarExpression strPredicate) {
```

Comment:
Ideally this should take `String` as inputs, like other `visitXXX` methods. But this method needs to patter match the predicate name and call `visitUnexpectedExpr`, which makes it a bit hard to take string input directly.

One proposal is to have 3 visit methods: `visitStartsWith` and so on.

Reply:
OK